lib/pages/settings/view/source_page.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:ui';
+
+import 'package:acs_upb_mobile/authentication/service/auth_provider.dart';
+import 'package:acs_upb_mobile/widgets/scaffold.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class SourcePage extends StatefulWidget {
+  static const String routeName = '/sources';
+
+  @override
+  _SourcePageState createState() => _SourcePageState();
+}
+
+class _SourcePageState extends State<SourcePage> {
+  bool wantsOfficialInfo;
+  bool wantsOrganizationInfo;
+  bool wantsStudentsInfo;
+  AuthProvider authProvider;
+
+  @override
+  void initState() {
+    super.initState();
+
+    authProvider = Provider.of<AuthProvider>(context, listen: false);
+    final user = authProvider.currentUserFromCache;
+    wantsOfficialInfo = user.sources?.contains('official') ?? false;
+    wantsOrganizationInfo = user.sources?.contains('organizations') ?? false;
+    wantsStudentsInfo = user.sources?.contains('students') ?? false;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AppScaffold(
+      actions: [
+        AppScaffoldAction(
+            text: S.current.buttonSave,
+            onPressed: () async {
+              final List<String> sources = ['official'];
+              if (wantsOrganizationInfo) sources.add('organizations');
+              if (wantsStudentsInfo) sources.add('students');
+              await authProvider.setSourcePreferences(sources);
+              Navigator.of(context).pop();
+            })
+      ],
+      title: Text(
+        S.current.sectionInformationSources,
+        style: const TextStyle(fontSize: 19.5),
+      ),
+      body: Column(
+        children: [
+          Padding(
+            padding: const EdgeInsets.all(10),
+            child: Text('${S.current.messageSelectSource}*'),
+          ),
+          Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 10),
+            child: Text(
+              S.current.hintSelectSources,
+              style: const TextStyle(fontWeight: FontWeight.bold),
+            ),
+          ),
+          const SizedBox(height: 10),
+          CheckboxListTile(
+            value: true,
+            onChanged: null,
+            title: Text(S.current.sourceOfficial),
+            subtitle: Text(S.current.sourceInfoOfficialWebPages),
+            controlAffinity: ListTileControlAffinity.leading,
+          ),
+          CheckboxListTile(
+            value: wantsOrganizationInfo,
+            onChanged: (value) {
+              setState(() {
+                wantsOrganizationInfo = value;
+              });
```

Comment:
Use lambda syntax to make this fit on one line, for readability. Same below.

Reply:
replaced this with lambda syntax